cuda_core/cuda/core/experimental/_module.py

```diff
@@ -168,7 +146,36 @@ def get_kernel(self, name):
             name = self._sym_map[name]
         except KeyError:
             name = name.encode()
+
+        self._lazy_load_module()
```

Comment:
Would it make sense to add `if self._handle is not None:` here? (In case `.get_kernel()` is called multiple times.)

In any case, I'd try to add a test like this:
```
oc = ObjectCode(...)
assert oc._handle is None
oc.get_kernel(...)
assert oc._handle is not None
```

Reply:
Yes I agree. @ksimpson-work please use `precondition` from `cuda.core.experimental._utils` (see how it's used in `_device.py`). We should do
```python
@precondition(_lazy_load_module)
def get_kernel(...):
    ...
```
and then check the handle availability in `_lazy_load_module` as Ralf suggested:
```python 
def _lazy_load_module(self):
    if self._handle is not None:
        return

    # ... init self._handle ...
```
This should be tested (see my comment below).